src/System.Composition.Runtime/tests/CompositionContractTests.cs

```diff
@@ -145,6 +145,44 @@ public static IEnumerable<object[]> Equals_TestData()
                 false
             };
 
+            yield return new object[]
+            {
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", new string[0] } }),
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", new object() } }),
+                false
+            };
+
+            yield return new object[]
+            {
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", null } }),
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", "value" } }),
+                false
+            };
+
+            yield return new object[]
+            {
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", "value" } }),
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", null } }),
+                false
+            };
+
+            yield return new object[]
+            {
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", new object[0] } }),
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", null } }),
+                false
+            };
+
+            if (!PlatformDetection.IsFullFramework)
```

Comment:
Is this necessary? Not sure if netfx uses the corefx code or not

Reply:
S.C.Runtime s a netstandard assembly so it doesn't harm to test it here.